### PR TITLE
Gitfs #11188 keichwa

### PR DESCRIPTION
**Pull request by keichwa**
These are the basics.
close https://github.com/SUSE/spacewalk/issues/11188

### CODE REVIEW

File: modules/salt/pages/salt-gitfs.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+Configuration options are set in the [path]``/etc/salt/master`` file or a separate configuration file [path]``/etc/salt/master.d/``.  Basic settings are:
+
+
+fileserver_backend::
+List of fileserver backends that the Salt master checks for files in the order they are defined.  Options:
++
+* [literal]``roots``: Files local on the Salt master ({productname} Server).
+{suse} strongly recommends to prefer [literal]``roots`` (local files) over [literal]``gitfs``.
```
Comment (mcalmer) [inline modules/salt/pages/salt-gitfs.adoc:15]: keeping `roots` is required to keep the product running. You can only "add" gitfs in addition. 
We should make it clear that it is not allowed to remove `roots`

**Comment by keichwa** [comment]
On 2020-05-04 11:35, Michael Calmer wrote:
> keeping `roots` is required to keep the product running. You can only
> "add" gitfs in addition.
> We should make it clear that it is not allowed to remove `roots`

Ok, thanks for feedback!

-- 
Karl Eichwalder                    SUSE Software Solutions Germany GmbH
                                    Maxfeldstraße 5
                                    90409 Nürnberg, Germany
Geschäftsführer: Felix Imendörffer (HRB 36809, AG Nürnberg)

File: modules/salt/pages/salt-gitfs.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
+[[salt.gitfs]]
+= The ``gitfs`` Fileserver Backend
+
+In {productname}, [package]``pygit2`` is the supported Python interface to git.
+When [package]``pygit2`` is installed the ``gitfs`` fileserver backend is available and it is a supported feature.
+
+
+Configuration options are set in the [path]``/etc/salt/master`` file or a separate configuration file [path]``/etc/salt/master.d/``.  Basic settings are:
```
Comment (Loquacity) [inline modules/salt/pages/salt-gitfs.adoc:8]:
```suggestion
Configuration options are set in the [path]``/etc/salt/master`` file, or in a separate configuration file called [path]``/etc/salt/master.d/``.  
The basic settings are:
```

File: modules/salt/pages/salt-gitfs.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+Example:
++
+----
+ext_pillar:
+  - git:
+    - master gitlab@gitlab.example.com:universe/setup.git:
+      - root: srv/pillar
+      - pubkey: /var/lib/salt/.ssh/id_rsa_gitlab.pub
+      - privkey: /var/lib/salt/.ssh/id_rsa_gitlab
+----
```
Comment (Loquacity) [inline modules/salt/pages/salt-gitfs.adoc:63]: Can we come up with a tidier IA here? Maybe a table?

Comment (keichwa) [reply modules/salt/pages/salt-gitfs.adoc:63]: I think simple lists with code examples are easier to read.  This is not the (new) reference guide ;)
[side note: writing adoc tables isn't fun.]

Comment (Loquacity) [reply modules/salt/pages/salt-gitfs.adoc:63]: I hate deflists more than I hate tables, I guess ;) I'll leave this in your capable hands, so do whatever you think works best.

Comment (keichwa) [reply modules/salt/pages/salt-gitfs.adoc:64]: ok, thanks!  if it helps we can later make use of the table style.

**Pull request merged**